components/suggest/src/yelp.rs

```diff
@@ -142,69 +153,187 @@ impl<'a> SuggestDao<'a> {
             return Ok(vec![]);
         }
 
-        let query_string = &query.keyword.trim();
-        if !query_string.contains(' ') {
-            let Some((subject, subject_exact_match)) = self.find_subject(query_string)? else {
-                return Ok(vec![]);
-            };
-            let (icon, icon_mimetype, score) = self.fetch_custom_details()?;
-            let builder = SuggestionBuilder {
-                subject: &subject,
-                subject_exact_match,
-                pre_modifier: None,
-                post_modifier: None,
-                location_sign: None,
-                location: None,
-                need_location: false,
-                icon,
-                icon_mimetype,
-                score,
-            };
-            return Ok(vec![builder.into()]);
-        }
+        let mut query_string = query.keyword.trim();
```

Comment:
Most of the functions that parse queries split with white space and process each one. It might be a waste if we split every time. So I want to try to split first, then parse it in each function.